### PR TITLE
Update dependencies on cardano-ledger-specs and ouroboros-network

### PR DESCRIPTION
**Pull request by mrBliss**

### CODE REVIEW

File: cardano-node/src/Cardano/Node/Protocol/Cardano.hs [hunk 1]
```diff
@@ -184,6 +185,13 @@ mkConsensusProtocolCardano NodeByronProtocolConfiguration {
            -- Alternatively, for testing we can transition at a specific epoch.
            --
            Just epochNo -> Consensus.TriggerHardForkAtEpoch epochNo)
+  where
+    -- The initial nonce, typically derived from the hash of Genesis config
+    -- JSON file.
+    --
+    -- WARNING: chains using different values of this parameter will be
+    -- mutually incompatible.
+    initialNonce = NeutralNonce
```
Comment (mrBliss) [inline cardano-node/src/Cardano/Node/Protocol/Cardano.hs:194]: This should be replaced with the hash of the Shelley genesis config JSON file, which should be taken as an argument/configuration parameter.

File: cardano-node/src/Cardano/Node/Protocol/Shelley.hs [hunk 2]
```diff
@@ -96,11 +96,18 @@ mkConsensusProtocolShelley NodeShelleyProtocolConfiguration {
...
         npcShelleyMaxSupportedProtocolVersion
         optionalLeaderCredentials
-
+  where
+    -- The initial nonce, typically derived from the hash of Genesis config
+    -- JSON file.
+    --
+    -- WARNING: chains using different values of this parameter will be
+    -- mutually incompatible.
+    initialNonce = NeutralNonce
```
Comment (mrBliss) [inline cardano-node/src/Cardano/Node/Protocol/Shelley.hs:110]: This should *also* be replaced with the hash of the Shelley genesis config JSON file, which should be taken as an argument/configuration parameter. But be careful, all testnets so far have used `NeutralNonce`, changing the value will require starting a new testnet.

**Comment by erikd** [comment]
Would be really good if the  `cardano-shell` udpate in https://github.com/input-output-hk/cardano-shell/pull/365 could be included in that.

**Comment by mrBliss** [comment]
> Would be really good if the `cardano-shell` udpate in [input-output-hk/cardano-shell#365](https://github.com/input-output-hk/cardano-shell/pull/365) could be included in that. I am just waiting for a CODEOWNERS approval.

Feel free to push an extra commit to this PR doing that. If that PR is only approved after your working hours, I'll try to remember to do it myself. You are not expecting breaking changes because of that PR, right?

**Comment by erikd** [comment]
There should not be any breaking changes.

File: cardano-node/src/Cardano/Node/Protocol/Cardano.hs (hunk 1 above)
Comment (dcoutts) [reply cardano-node/src/Cardano/Node/Protocol/Cardano.hs:194]: Can we add a TODO?
```suggestion
    -- TODO: This should be replaced with the hash of the Shelley genesis config
    -- JSON file, which should be taken as an argument/configuration parameter.
    initialNonce = NeutralNonce
```

File: cardano-node/src/Cardano/Node/Protocol/Shelley.hs (hunk 2 above)
Comment (dcoutts) [reply cardano-node/src/Cardano/Node/Protocol/Shelley.hs:110]:
```suggestion
    -- TODO: This should also be replaced with the hash of the Shelley genesis config
    -- JSON file, which should be taken as an argument/configuration parameter.
    -- But be careful, all testnets so far have used NeutralNonce, changing the value
    -- will require starting a new testnet.
    initialNonce = NeutralNonce
```

**Comment by erikd** [comment]
https://github.com/input-output-hk/cardano-shell/pull/365 has been merged.

**Comment by mrBliss** [comment]
bors merge

**Pull request merged**